### ISSUE TITLE
[Bug][iOS] Clarity SDK Crash: NSInvalidArgumentException 

### ISSUE THREAD

**Issue by hamed-ice** [microsoft/clarity-apps#78]
### Clarity Bug Report

---

**Describe the Bug**  
The app crashes with an **NSInvalidArgumentException**, as reported in Firebase. This crash seems is related to the Clarity SDK and has been present since our app version 1.2.4. in this version, we only released Clarity

---

**Repro Steps**  
Unable to reproduce the crash reliably. It appears to occur in some user environments without a consistent pattern.  

---

**Expected Behavior**  
The application should function normally without crashing, with the Clarity SDK integrated.  

---

**Screenshots**  
Not applicable as the issue occurs as a crash with no specific UI interaction.  

---

**Project Info**  
If required, the **Project ID** will be shared with `clarity-apps-support@microsoft.com` along with the issue number in the subject line.  

---

**Platform/Framework Info**  
- **Device**: Various, including iPhone models  
- **OS**: iOS SDK > 15
- **Application Framework**: Native UIKit-based  
- **Clarity Version**: Unknown, likely the version included in app version 1.2.4  

---

**Additional Context**  
The crash stack trace indicates the issue occurs during animation handling, specifically related to the `CA_interpolateValue` method in QuartzCore and a missing UUID in the Clarity SDK.  

**Firebase Crash Report**  
```
    Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x83f20 __exceptionPreprocess
1  libobjc.A.dylib                0x172b8 objc_exception_throw
2  CoreFoundation                 0x18d480 +[NSObject(NSObject) _copyDescription]
3  CoreFoundation                 0x20fb4 ___forwarding___
4  CoreFoundation                 0x208d0 _CF_forwarding_prep_0
5  QuartzCore                     0xb4264 -[NSNumber(CAAnimatableValue) CA_interpolateValue:byFraction:]
6  QuartzCore                     0xb0b50 -[CABasicAnimation applyForTime:presentationObject:modelObject:]
7  QuartzCore                     0xc3878 CA::Layer::presentation_layer(CA::Transaction*)
8  Clarity                        0xc9524 (Missing UUID 0370cc1d---
9  Clarity                        0xc9e3c (Missing UUID 0370cc1d
10 Clarity                        0xc9e3c (Missing UUID 0370cc1d
11 Clarity                        0xc9e3c (Missing UUID 0370cc1d
12 Clarity                        0xc9e3c (Missing UUID 0370cc1d7
13 Clarity                        0xc9e3c (Missing UUID 0370cc1d777
14 Clarity                        0xc8ee4 (Missing UUID 0370cc1d7770379aa5a1d8e
15 Clarity                        0xc8b24 (Missing UUID 0370cc1d7770379aa5a1d8
16 Clarity                        0xf44d8 (Missing UUID 0370cc1d7770379aa5a1d8e
17 Clarity                        0xf3f98 (Missing UUID 0370cc1d7770379aa5a1d8eb
18 Clarity                        0xa1b18 (Missing UUID 0370cc1d7770379aa5a1d8eb
19 libdispatch.dylib              0x3dd4 _dispatch_client_callout
20 libdispatch.dylib              0x72d8 _dispatch_continuation_pop
21 libdispatch.dylib              0x1b1c8 _dispatch_source_latch_and_call
22 libdispatch.dylib              0x19d8c _dispatch_source_invoke
23 libdispatch.dylib              0x124b4 _dispatch_main_queue_drain
24 libdispatch.dylib              0x121b8 _dispatch_main_queue_callback_4CF
25 CoreFoundation                 0x56710 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
26 CoreFoundation                 0x53914 __CFRunLoopRun
27 CoreFoundation                 0x52cd8 CFRunLoopRunSpecific
28 GraphicsServices               0x11a8 GSEventRunModal
29 UIKitCore                      0x40aae8 -[UIApplication _run]
30 UIKitCore                      0x4bed98 UIApplicationMain
31 Fedshi                         0x80a4 main + 21 (main.swift:21)
32 ???                            0x1c1eef154 (Missing)
        
...  
```

**Labels**: Crash, NSInvalidArgumentException, Clarity SDK, UIKit  

--- 

Let me know if you need further details or clarification.

**Comment by ibradwan** [comment]
Hi @hamed-ice,

This issue should have been resolved in iOS SDK version 2.2.2. Please upgrade and let us know if you have any feedback.

**Comment by hamed-ice** [comment]
> Hi [@hamed-ice](https://github.com/hamed-ice),
> 
> This issue should have been resolved in iOS SDK version 2.2.2. Please upgrade and let us know if you have any feedback.

Hi @ibradwan , I'll update it and let you know the results after the new release.
